ledger/participant-integration-api/src/main/scala/platform/apiserver/meteringreport/HmacSha256.scala

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.apiserver.meteringreport
+
+import spray.json.DefaultJsonProtocol._
+import spray.json.RootJsonFormat
+
+import java.util.Base64
+import javax.crypto.spec.SecretKeySpec
+import javax.crypto.{KeyGenerator, Mac}
+import scala.util.Try
+
+object HmacSha256 {
+
+  def toBase64(bytes: Array[Byte]): String = Base64.getUrlEncoder.encodeToString(bytes)
+
+  // The key used for both mac and key generation as defined in
+  // https://docs.oracle.com/javase/9/docs/specs/security/standard-names.html
+  val algorithm = "HmacSHA256"
+
+  def compute(key: Key, message: Array[Byte]): Either[Throwable, Array[Byte]] = {
```

Comment:
How about exposing a base64 encoded string here, as anyway you convert it further later?

Reply:
I would not expose `toBase64` as public def in general

Reply:
This specific method though is exposing pure Array[Byte] which is then transformed to base64 encoded string, I am challenging the need of exposing Array[Byte] as anyway there is no use case you would ever use Array[Byte] directly without base64 transformation

Reply:
or I just miss a usage which is not in the code yet :)

Reply:
I think for serialization `Base64` is good but for internal representation Array[Byte] is better.  The main reason for this is that the transform to base 64 string to Array[Byte] may throw an exception.